components/schemes/Color/tests/unit/MDCSemanticColorSchemeTests.m

```diff
@@ -67,4 +67,58 @@ - (void)testInitWithMaterialDefaults {
   XCTAssertEqualObjects(colorScheme.onBackgroundColor, ColorFromRGB(0x000000));
 }
 
+- (void)testColorMergeForOpaqueColor {
+  UIColor *backgroundColor = [UIColor whiteColor];
+  UIColor *blendColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:1.0];
+  UIColor *expectedColor =
+      [UIColor colorWithRed:0.f green:0.f blue:0.f alpha:1.0f];
+  UIColor *resultColor =
+      [MDCSemanticColorScheme blendColor:blendColor withBackgroundColor:backgroundColor];
+  XCTAssertEqualObjects(resultColor, expectedColor);
+}
+
+- (void)testColorMergeFor50OpacityBlackOnWhite {
+  UIColor *backgroundColor = [UIColor whiteColor];
+  UIColor *blendColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:0.5];
+  UIColor *expectedColor =
+      [UIColor colorWithRed:0.5f green:0.5f blue:0.5f alpha:1.0f];
+  UIColor *resultColor =
+      [MDCSemanticColorScheme blendColor:blendColor withBackgroundColor:backgroundColor];
+  XCTAssertEqualObjects(resultColor, expectedColor);
+}
+
+- (void)testColorMergeFor60GrayOpacityOnWhite {
+  UIColor *backgroundColor = [UIColor whiteColor];
+  UIColor *blendColor =
+      [UIColor colorWithRed:230.0f/255.0f green:230.0f/255.0f blue:230.0f/255.0f alpha:0.60f];
+  UIColor *resultColor =
+      [MDCSemanticColorScheme blendColor:blendColor withBackgroundColor:backgroundColor];
+  UIColor *expectedColor =
+      [UIColor colorWithRed:240.0f/255.0f green:240.0f/255.0f blue:240.0f/255.0f alpha:1.0f];
```

Comment:
Please add a test with non-uniform color channel information to detect if the color channels get mixed by accident.

Reply:
Done.